### PR TITLE
Add support for casper 1.0 style of arguments.

### PR DESCRIPTION
**Pull request by francois2metz**
See #68.

This a WIP to support casper 1.0 arguments style.

I added support for multiple arguments to the `thenEvaluate` method.

The `allowMore` is quite hacky and need some work. Any thought?

### CODE REVIEW

File: tests/test/then.js [hunk 1]
```diff
@@ -148,13 +148,14 @@ describe("Spooky provides Casper's then* functions", function () {
             function (done) {
                 context.spooky.start(FIXTURE_URL + '/1.html');
 
-                context.spooky.thenEvaluate(function (foo) {
+                context.spooky.thenEvaluate(function (foo, bar) {
                     window.__foo = foo;
-                }, { foo: 'bar' });
+                    window.__bar = bar;
+                }, 'foo', 'bar');
```
Comment (lawnsea) [inline tests/test/then.js]: This removes the test for the old style of passing args. I don't know if that's what we want to do.

File: lib/bootstrap/casper.js [hunk 2]
```diff
@@ -56,11 +56,12 @@ var methodsToProvide = {
     withFrame: ['frameInfo', function then () {}],
     withPopup: ['popupInfo', function then () {}]
 };
+methodsToProvide.thenEvaluate.allowMore = true;
```
Comment (lawnsea) [inline lib/bootstrap/casper.js]: I wonder if we could instead append `...` to the parameter name of any variadic arguments:

```javascript
thenEvaluate: [function fn () {}, 'replacements...']
```

**Comment by lawnsea** [comment]
I think it's going to be tricky to make pre-1.0 and 1.0 work together. I'd rather see the Casper version added to the options object. `lib/bootstrap/casper` could then set up the appropriate RPC methods for the passed version.

File: lib/bootstrap/casper.js (hunk 2 above)
Comment (francois2metz) [inline lib/bootstrap/casper.js]: Nice idea. Will try that.

File: tests/test/then.js (hunk 1 above)
Comment (francois2metz) [inline tests/test/then.js]: I may keep the original test around. I was just lazy :flushed:

Comment (lawnsea) [inline tests/test/then.js]: Heh, ok. :)

**Comment by francois2metz** [comment]
I updated the pull request with your comments.

I also tried to detect casperjs < 1.0.0-RC5 to update the rpc method signature.
There is already failed test on pre 1.0.0 because of some undefined methods (`waitForPopup` and `withPopup` for instance).  See https://travis-ci.org/francois2metz/SpookyJS/builds/59798131

I expect some pain, but we can disable these tests in old version of casperjs.

I don't know what to do. Is supporting casper < 1.0.0-RC5 required?

**Comment by lawnsea** [comment]
Ok, great! I'll take a look this weekend.

> I don't know what to do. Is supporting casper < 1.0.0-RC5 required?

I'm not sure. I don't use Spooky much these days, so I think I'd like to hear from the users. I suppose we could just dodge the question with a bit of semver.

**Pull request closed without merging**